custom_components/nordpool/__init__.py

```diff
@@ -108,12 +110,29 @@ async def _someday(self, area: str, currency: str, day: str):
     async def today(self, area: str, currency: str) -> dict:
         """Returns today's prices in an area in the requested currency"""
         return await self._someday(area, currency, "today")
-        
 
     async def tomorrow(self, area: str, currency: str):
         """Returns tomorrow's prices in an area in the requested currency"""
         return await self._someday(area, currency, "tomorrow")
-        
+
+    def tomorrow_valid(self, area: str, currency: str) -> bool:
```

Comment:
Can you move all the validation to aio_price? I would prefer to do it there, this allows for easier testing.

Reply:
That is indeed what I initially tried to do, before you explained to me why what I was doing isn't possible.

Since AioPrices doesn't keep a copy of the data (it only retrieves it, the result is stored by NordpoolData) the approach I ended up with doesn't work there.

A combination of the two variants I have so far attempted is probably possible by having AioPrices raise an exception on invalid data and handling that exception in NordpoolData (instead of having AioPrices retry like I had previously).

Reply:
I have now tested and pushed an implementation of that idea.